src/browser/NativeMessageInstaller.cpp

```diff
@@ -0,0 +1,313 @@
+/*
+ *  Copyright (C) 2017 Sami Vänttinen <sami.vanttinen@protonmail.com>
+ *  Copyright (C) 2017 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "NativeMessageInstaller.h"
+#include "BrowserSettings.h"
+#include "config-keepassx.h"
+
+#include <QCoreApplication>
+#include <QDebug>
+#include <QDir>
+#include <QFile>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QMessageBox>
+#include <QProcessEnvironment>
+#include <QSettings>
+#include <QStandardPaths>
+
+using namespace BrowserShared;
+
+namespace
+{
+    const QString HOST_NAME = QStringLiteral("org.keepassxc.keepassxc_browser");
```

Comment:
I suppose this couldn't be changed to `org.keepassxc.KeePassXC.KeePassXCBrowser` without breaking stuff, could it?

Reply:
We cannot change this. From the specification: "This name can only contain lowercase alphanumeric characters, underscores and dots. The name cannot start or end with a dot, and a dot cannot be followed by another dot."

Reply:
Should be org.keepassxc.keepassxc.browser then, unless it breaks something, in that case just leave it as is.

Reply:
Leave it as it is.